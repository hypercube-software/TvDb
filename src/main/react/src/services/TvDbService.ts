import {Response} from "../model/Response";
import SearchResult from "../model/SearchResult";
import TvShow from "../model/TvShow";

class TvDbService {
    delete(tvShowId: number):Promise<Response> {
        const payload = {
            method: 'DELETE'
        }
        return fetch('/api/tv/delete?id=' + tvShowId, payload)
            .then(r => r.json());
    }

    create(tvShow: TvShow):Promise<Response> {
        const payload = {
            method: 'POST',
            body: JSON.stringify(tvShow),
            headers: {
                'Content-type': 'application/json; charset=UTF-8'
            }
        }
        return fetch('/api/tv/create', payload)
            .then(r => r.json());
    }

    update(tvShow: TvShow):Promise<Response> {
        const payload = {
            method: 'PUT',
            body: JSON.stringify(tvShow),
            headers: {
                'Content-type': 'application/json; charset=UTF-8'
            }
        }
        return fetch('/api/tv/update', payload)
            .then(r => r.json());
    }

    search(query: string, externalSearch: boolean):Promise<SearchResult> {        
        return fetch('/api/tv/search?q=' + query + "&ext=" + externalSearch)
            .then(r => r.json());
    }
}

export default new TvDbService();