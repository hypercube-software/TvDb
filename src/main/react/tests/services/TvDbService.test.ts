import TvDbService from '../../src/services/TvDbService';
const nodeFetch = require("node-fetch");
function myFetch(url: string): Promise<Response> {
    return nodeFetch("http://localhost:8080" + url);
}
global.fetch = myFetch;

describe('TvDbService tests', () => {

    it("can do an external search", () => {
        return expect(TvDbService.search("The A-Team", true).then(sr => sr.result.filter(tvShow => {
            return tvShow.name === "The A-Team";
        }))).resolves.toMatchSnapshot();
    })

});