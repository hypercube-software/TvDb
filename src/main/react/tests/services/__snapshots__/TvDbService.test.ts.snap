// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvDbService tests can do an external search 1`] = `
Array [
  Object {
    "artwork": "http://static.tvmaze.com/uploads/images/original_untouched/6/17013.jpg",
    "id": null,
    "name": "The A-Team",
    "summary": "<p><b>The A-Team</b> is an American action-adventure television series, running from 1983 to 1987, about a fictional group of exâ€“United States Army Special Forces.</p>",
    "thumbnail": "http://static.tvmaze.com/uploads/images/medium_portrait/6/17013.jpg",
    "url": "http://www.tvmaze.com/shows/1058/the-a-team",
  },
]
`;
